File: 3705-find-the-largest-almost-missing-integer/find-the-largest-almost-missing-integer.cpp
class Solution {
public:
    int largestInteger(vector<int>& nums, int k) {
        int n = nums.size();
        unordered_map<int, int> mp;

        for (int i = 0; i <= n - k; ++i) {
            unordered_set<int> st(nums.begin() + i, nums.begin() + i + k);
            for (int num : st) {
                mp[num]++;
            }
        }

        int ans = -1;
        for (const auto& [num, count] : mp) {
            if (count == 1 && num > ans) {
                ans = num;
            }
        }

        return ans;
    }
};

File: 3705-find-the-largest-almost-missing-integer/find-the-largest-almost-missing-integer.py
class Solution:
    def largestInteger(self, nums: List[int], k: int) -> int:
        n = len(nums)
        freq = {}

        for i in range(n - k + 1):
            for num in set(nums[i : i + k]):
                freq[num] = freq.get(num, 0) + 1

        ans = -1
        for num, count in freq.items():
            if count == 1 and num > ans:
                ans = num

        return ans
